core/cloudflow-operator/src/main/scala/cloudflow/operator/event/ConfigInputChangeEvent.scala

```diff
@@ -236,16 +241,28 @@ object ConfigInputChangeEvent {
     // TODO implement
     config
 
-  def moveTopicsConfigToPortMappings(app: CloudflowApplication.CR, config: Config, streamletName: String): Config =
-    // convert to cloudflow.runner.streamlet.port_mappings.topic
-    // TODO does merge work well on arrays? (port mappings / connected ports is an array, maybe change to 'map')
-    // TODO implement
-    config
+  // move cloudflow.topics.<topic> config to cloudflow.runner.streamlet.context.port_mappings.<port>.config
+  // and let the merge magic happen!
+  def moveTopicsConfigToPortMappings(deployment: cloudflow.blueprint.deployment.StreamletDeployment,
+                                     streamletConfig: Config,
+                                     appConfig: Config,
+                                     streamletName: String): Config = {
+    val portMappingConfigs = deployment.portMappings.flatMap {
```

Comment:
Didn't we rename port mappings elsewhere?

Reply:
it is now called port mappings